src/langcheck/metrics/eval_clients/_gemini.py

```diff
@@ -0,0 +1,242 @@
+from __future__ import annotations
+
+import asyncio
+import os
+from typing import Any, Iterable
+
+import google.ai.generativelanguage as glm
+import google.generativeai as genai
+import torch
+
+from langcheck.utils.progess_bar import tqdm_wrapper
+
+from ..prompts._utils import get_template
+from ..scorer._base import BaseSimilarityScorer
+from ._base import EvalClient
+
+
+class GeminiEvalClient(EvalClient):
+    '''EvalClient defined for the Gemini model.
+    '''
+
+    def __init__(self,
+                 model: genai.GenerativeModel | None = None,
+                 model_args: dict[str, Any] | None = None,
+                 generate_content_args: dict[str, Any] | None = None,
+                 embed_model_name: str | None = None):
+        '''
+        Initialize the Gemini evaluation client. The authentication
+        information is automatically read from the environment variables,
+        so please make sure GOOGLE_API_KEY is set.
+
+        Ref:
+            https://ai.google.dev/api/python/google/generativeai/GenerativeModel
+
+        Args:
+            model: (Optional) The Gemini model to use. If not provided, the
+                model will be created using the model_args.
+            model_args: (Optional) Dict of args to create the Gemini model.
+            generate_content_args: (Optional) Dict of args to pass in to the
+                ``generate_content`` function.
+            embed_model_name: (Optional) The name of the embedding model to use.
+                If not provided, the models/embedding-001 model will be used.
+        '''
+        if model:
+            self._model = model
+        else:
+            genai.configure(api_key=os.getenv('GOOGLE_API_KEY'))
+            model_args = model_args or {}
+            self._model = genai.GenerativeModel(**model_args)
+
+        self._generate_content_args = generate_content_args or {}
+        self._embed_model_name = embed_model_name
+
+        # TODO: Allow the user to specify the use of async. There currently
+        # seems to be an issue with `generate_content_async()` that is blocking
+        # this: https://github.com/google-gemini/generative-ai-python/issues/207
+        self._use_async = False
+
+    def _call_api(self,
+                  prompts: Iterable[str | None],
+                  config: dict[str, Any],
+                  *,
+                  tqdm_description: str | None = None) -> list[Any]:
+        # A helper function to call the API with exception filter for alignment
+        # of exception handling with the async version.
+        def _call_api_with_exception_filter(prompt: str) -> Any:
+            try:
+                return self._model.generate_content(prompt, **config)
+            except Exception as e:
+                return e
+
+        if self._use_async:
```

Comment:
If the branch is not used, I think it is less confusing to comment the lines out or just remove them.